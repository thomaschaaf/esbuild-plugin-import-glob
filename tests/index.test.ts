import { build } from 'esbuild';
import requireContext from '../src/index';
import path from 'path';

describe('test', () => {
  it('should', async () => {
    const logs = await runTestFile('service/main.ts');

    const migrationModules = logs[0][0];
    expect(migrationModules).toMatchSnapshot();

    const entitiesModules = logs[1][0];
    expect(entitiesModules).toMatchSnapshot();

    const migration2Modules = logs[2][0];
    expect(migrationModules.default).toEqual(migration2Modules);
  });

  it('keeps proper scope of imports', async () => {
    const logs = await runTestFile('conflicts/main.ts');

    const [setA, setB] = logs[0];
    expect(setA).toMatchSnapshot();
    expect(setB).toMatchSnapshot();
  });

  async function runTestFile(testFile) {
    const result = await build({
      entryPoints: [path.resolve(__dirname, testFile)],
      write: false,
      plugins: [requireContext()],
      outfile: `tests/bundle.js`,
      bundle: true,
    });

    const fakeLogger = jest.fn();

    eval(`(console) => ${result.outputFiles[0].text}`)({ log: fakeLogger });

    return fakeLogger.mock.calls;
  }
});
