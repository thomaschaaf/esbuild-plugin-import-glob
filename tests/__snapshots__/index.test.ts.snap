// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test keeps proper scope of imports 1`] = `
Object {
  "default": Object {
    "default": Array [
      Object {
        "default": "a",
      },
      Object {
        "default": "b",
      },
    ],
    "filenames": Array [
      "./cases/a.js",
      "./cases/b.js",
    ],
  },
}
`;

exports[`test keeps proper scope of imports 2`] = `
Object {
  "default": Object {
    "default": Array [
      Object {
        "default": "c",
      },
      Object {
        "default": "d",
      },
    ],
    "filenames": Array [
      "./cases/c.js",
      "./cases/d.js",
    ],
  },
}
`;

exports[`test should 1`] = `
Object {
  "default": Array [
    Object {
      "default": "1js",
    },
    Object {
      "default": "2ts",
    },
    Object {
      "default": "3ts",
      "test": "test3ts",
    },
    Object {
      "default": Object {},
    },
    Object {
      "noDefaultExport": "subfolder",
    },
  ],
  "filenames": Array [
    "./migrations/01.js",
    "./migrations/02.ts",
    "./migrations/03.ts",
    "./migrations/10.js",
    "./migrations/subfolder/sub.js",
  ],
}
`;

exports[`test should 2`] = `
Object {
  "default": Array [
    Object {
      "default": "Server",
    },
    Object {
      "default": "User",
    },
  ],
  "filenames": Array [
    "./entities/server.ts",
    "./entities/user.ts",
  ],
}
`;
